applications/crossbar/src/crossbar_maintenance.erl

```diff
@@ -413,20 +404,91 @@ create_account(AccountName, Realm, Username, Password) ->
                   ,kz_term:to_binary(Password)
                   ).
 
+maybe_promote_account(Context) ->
+    AccountDb = cb_context:account_db(Context),
+    AccountId = cb_context:account_id(Context),
+
+    case kapps_util:get_all_accounts() of
+        [AccountDb] ->
+            'ok' = promote_account(AccountId),
+            'ok' = allow_account_number_additions(AccountId),
+            'ok' = whs_account_conversion:force_promote(AccountId),
+            update_system_config(AccountId);
+        _Else -> 'ok'
+    end.
+
+-spec create_account_and_user(kz_json:object(), kz_json:object()) ->
+                                     {'ok', cb_context:context()}.
+create_account_and_user(Account, User) ->
+    Funs = [fun prechecks/1
+           ,{fun validate_account/2, Account}
+           ,fun create_account/1
+           ,{fun validate_user/2, User}
+           ,fun create_user/1
+           ],
+    lists:foldl(fun create_fold/2
+               ,{'ok', cb_context:new()}
+               ,Funs
+               ).
+
+-spec create_fold(fun() | {fun(), kz_json:object()}, {'ok', cb_context:context()}) ->
+                         {'ok', cb_context:context()}.
+create_fold({F, V}, {'ok', C}) -> F(V, C);
+create_fold(F, {'ok', C}) -> F(C).
+
 -spec update_system_config(ne_binary()) -> 'ok'.
 update_system_config(AccountId) ->
     kapps_config:set(?KZ_SYSTEM_CONFIG_ACCOUNT, <<"master_account_id">>, AccountId),
     io:format("updated master account id in system_config.~s~n", [?KZ_SYSTEM_CONFIG_ACCOUNT]).
 
+-spec prechecks(cb_context:context()) -> {'ok', cb_context:context()}.
+prechecks(Context) ->
+    Funs = [fun db_accounts_exists/0
+           ,fun db_system_config_exists/0
+           ,fun db_system_schemas_exists/0
+           ],
+    'true' = lists:all(fun(F) -> F() end, Funs),
+    {'ok', Context}.
```

Comment:
So this is the only thing that PR adds, right? It is fixing the race condition that CircleCI found. Was there something else that changed in this PR?